src/input_xml.F90

```diff
@@ -2818,6 +2935,207 @@ subroutine read_tallies_xml()
+        allocate(t % score(t % n_user_triggers))
+        ! Get the scores for each trigger
+        Trigger_Loop: do trig_ind =1, t % n_user_triggers
+
+        ! Get pointer to trigger mode
+          call get_list_item(node_trigger_list, trig_ind , node_trigger)
+
+          ! Get scores information
+          if (check_for_node(node_trigger, "scores")) then
+            call get_node_value(node_trigger, "scores", score_name)
+            do imomstr = 1, size(MOMENT_STRS)
+              if (starts_with(score_name,trim(MOMENT_STRS(imomstr)))) then
+                n_order_pos = scan(score_name,'0123456789')
+                n_order = int(str_to_int( &
+                score_name(n_order_pos:(len_trim(score_name)))),4)
+                if (n_order > MAX_ANG_ORDER) then
+                  ! User requested too many orders; throw a warning and
+                  ! set to the maximum order.
+                  ! The above scheme will essentially take the absolute value
+                  call warning("Invalid scattering order of " &
+                      // trim(to_str(n_order)) // " requested. Setting to the &
+                      & maximum permissible value, " &
+                      // trim(to_str(MAX_ANG_ORDER)))
+                  n_order = MAX_ANG_ORDER
+                end if
+                score_name = trim(MOMENT_STRS(imomstr)) // "n"
+                ! Find total number of bins for this case
+                if (imomstr >= YN_LOC) then
+                  n_bins = (n_order + 1)**2
+                else
+                  n_bins = n_order + 1
+                end if
+                exit
+              end if
+            end do
+            ! Check the Moment_N_Strs, but only if we werent successful above
+            if (imomstr > size(MOMENT_STRS)) then
+              do imomstr = 1, size(MOMENT_N_STRS)
+                if (starts_with(score_name,trim(MOMENT_N_STRS(imomstr)))) then
+                  n_order_pos = scan(score_name,'0123456789')
+                  n_order = int(str_to_int( &
+                  score_name(n_order_pos:(len_trim(score_name)))),4)
+                  if (n_order > MAX_ANG_ORDER) then
+                    ! User requested too many orders; throw a warning and set to
+                    ! the maximum order.
+                    ! The above scheme will essentially take the absolute value
+                    call warning("Invalid scattering order of " &
+                         // trim(to_str(n_order)) // " requested. Setting to &
+                         & the maximum permissible value, " &
+                         // trim(to_str(MAX_ANG_ORDER)))
+                    n_order = MAX_ANG_ORDER
+                  end if
+                  score_name = trim(MOMENT_N_STRS(imomstr)) // "n"
+                  exit
+                end if
+              end do
+            end if
+            t % score(trig_ind) % score_name = score_name
+
+            select case (trim(score_name))
```

Comment:
Do we need this `select case` block at all?  It looks like we may be able to replace it with something like `t % score(trig_ind) % position = find_score % get_key(t % score_bins(j))`